tools/github-issues/Azure.Sdk.Tools.GitHubIssues/Helpers/GitHubHelpers.cs

```diff
@@ -8,6 +11,13 @@ namespace GitHubIssues.Helpers
 {
     internal static class GitHubHelpers
     {
+        private static TimeLimiter rateLimiter = TimeLimiter.GetFromMaxCountByInterval(1, TimeSpan.FromSeconds(1));
+
+        private static void WaitForGitHubCapacity()
+        {
+            rateLimiter.GetAwaiter().GetResult();
```

Comment:
How is this different than doing a 'await Task.Delay(1000)' before each call to GitHub?

Reply:
In the case of GitHub Issues, given it is mostly sync code, probably not much. But I thought I would test it out here since the rate limiting issue in GitHub issues is easily reproducible in the prod environment. And assuming it works I could lift the same solution in for Check Enforcer.

Reply:
The other thing is that Task.Delay(...) guarantees that you get a delay. But in reality we only want to hold off making a request when the rate is exceeded. So its possible that depending on the data volumes, serialization work, mail sending etc that you might be able to do two requests without an explicit delay.

Reply:
I thought we were trying to ensure that we only do x requests per minute, and we could simulate that with a simple `Delay`.

I actually think that we should make all our tools do that as much as possible because otherwise you can have a tool that will blow through the rate limit for that account and that would block all other apps from running. So instead of stepping back when you hit the limit you would preemptively try to not reach the limit.... if possible.